source/MaterialXTest/GenMdl.cpp

```diff
@@ -109,6 +109,52 @@ TEST_CASE("GenShader: MDL Unique Names", "[genmdl]")
 }
 */
 
+void MdlShaderGeneratorTester::compileSource(const std::vector<mx::FilePath>& sourceCodePaths)
+{
+    if (sourceCodePaths.empty() || sourceCodePaths[0].isEmpty())
+        return;
+    
+    std::string mdlcExec(MATERIALX_MDLC_EXECUTABLE);
+    if (mdlcExec.empty())
+    {
+        return;
+    }
+
+    mx::FilePath moduleToTestPath = sourceCodePaths[0].getParentPath();
+    mx::FilePath module = sourceCodePaths[0];
+    std::string moduleToTest = module[module.size()-1];
+    moduleToTest = moduleToTest.substr(0, moduleToTest.size() - sourceCodePaths[0].getExtension().length() - 1);
+
+    mx::StringVec extraModulePaths = mx::splitString(MATERIALX_MDL_MODULE_PATHS, ",");
+
+    std::string mdlcCommand = mdlcExec;
+    for (const std::string& extraPath : extraModulePaths)
+    {
+        mdlcCommand += " -p\"" + extraPath + "\"";
+    }
+    mx::FilePath currentPath = mx::FilePath::getCurrentPath();
+    mx::FilePath coreModulePath = currentPath / mx::FilePath("libraries/stdlib/genmdl");
```

Comment:
Hard coded this for now. If this is not the location then you need to set MATERIALX_MDL_MODULE_PATHS to point to the module paths required.

Reply:
I still don't know how to get rid of the error coming from mdlc for missing filenames on image lookups (mentioned).
So still getting things like this:

_Error: d:\Work\materialx\MaterialX-adsk-public-master\build_public_rel\source\MaterialXTest\generatedshaders\genmdl\mdl\image4_output.mdl(17,49): Error C298: Unable to resolve file path './resources/Images/grid.png'
	Error: d:\Work\materialx\MaterialX-adsk-public-master\build_public_rel\source\MaterialXTest\generatedshaders\genmdl\mdl\image4_output.mdl(17,49): Note: did you mean '/resources/Images/grid.png'?
	Error: D:/Work/materialx/mdl-sdk-325000.1814/nt-x86-64/bin/mdlc: 1 errors detected in module image4_output_

Reply:
Maybe we could avoid emitting the file path value for the parameter, to avoid this.

We could introduce a generator option for emitting default values or not. This is something the WebGL team also wants I think, since default values on input uniforms was not supported by GLSL-ES (If I understood them correctly).